### PR TITLE
[BugFix]fixed rm_router_logits_allgather_ep bug

### PR DESCRIPTION
**Pull request by ttanzhiqiang**
There are four situations in the previous logic
1. If Prefill/decode uses AllGather or NaiveMulticast at the same time, this logic is normal, and this solution is used for optimization
2. If Prefill/decode uses All2All/MC2 at the same time, this logic is also normal, and this solution is not used for optimization
3. Prefill uses AllGatherEP (using VLLM_ENABLE_FUSED_EXPERTS_ALLGATHER_EP switch) and Decode uses MC2, which will affect the results. There is a bug in this place
4. In the PD separation scenario, the strategies used by P and D are separate, so there is no impact.

rm_router_logits optimization scheme, AllGather/NaiveMulticast/All2All/MC2 are all used
1. If Prefill/decode use AllGather or NaiveMulticast scheme at the same time, this logic is normal, and this scheme is used for optimization
2. If Prefill/decode use All2All/MC2 scheme at the same time, this logic is also normal, and this scheme is used for optimization
3. Prefill uses AllGatherEP scheme (use VLLM_ENABLE_FUSED_EXPERTS_ALLGATHER_EP switch), Decode uses MC2 scheme, and this scheme is used for optimization
4. In the PD separation scenario, the strategies used by P and D are separate, and this scheme is used for optimization.

#### Test method for case 1
export VLLM_USE_V1=1
export TASK_QUEUE_ENABLE=1
source /usr/local/Ascend/ascend-toolkit/set_env.sh
source /usr/local/Ascend/nnal/atb/set_env.sh
export ASCEND_LAUNCH_BLOCKING=0
export VLLM_VERSION=0.9.1
nohup python -m vllm.entrypoints.openai.api_server --model=/mnt/deepseek/DeepSeek-R1-W8A8-VLLM \
    --served-model-name auto \
    --quantization ascend \
    --trust-remote-code \
    --distributed-executor-backend=mp \
    --port 8006 \
    -tp=4 \
    -dp=4 \
    --max-num-seqs 24 \
    --max-model-len 32768 \
    --max-num-batched-tokens 32768 \
    --block-size 128 \
    --no-enable-prefix-caching \
    --additional-config '{"torchair_graph_config":{"enabled":true,"use_cached_graph":true,"graph_batch_sizes":[24]},"ascend_scheduler_config":{"enabled":true},"expert_tensor_parallel_size":16}' \
    --gpu-memory-utilization 0.96 &> run.log &
disown

#### Test method for case 2
export VLLM_USE_V1=1
export TASK_QUEUE_ENABLE=1
source /usr/local/Ascend/ascend-toolkit/set_env.sh
source /usr/local/Ascend/nnal/atb/set_env.sh
export ASCEND_LAUNCH_BLOCKING=0
export VLLM_VERSION=0.9.1
nohup python -m vllm.entrypoints.openai.api_server --model=/mnt/deepseek/DeepSeek-R1-W8A8-VLLM \
    --served-model-name auto \
    --quantization ascend \
    --trust-remote-code \
    --distributed-executor-backend=mp \
    --port 8006 \
    -tp=4 \
    -dp=4 \
    --max-num-seqs 24 \
    --max-model-len 32768 \
    --max-num-batched-tokens 32768 \
    --block-size 128 \
    --no-enable-prefix-caching \
    --enable_expert_parallel \
    --additional-config '{"torchair_graph_config":{"enabled":true,"use_cached_graph":true,"graph_batch_sizes":[24]},"ascend_scheduler_config":{"enabled":true},"expert_tensor_parallel_size":16}' \
    --gpu-memory-utilization 0.96 &> run.log &
disown

#### Test method for case 3
export VLLM_USE_V1=1
export TASK_QUEUE_ENABLE=1
source /usr/local/Ascend/ascend-toolkit/set_env.sh
source /usr/local/Ascend/nnal/atb/set_env.sh
export ASCEND_LAUNCH_BLOCKING=0
export VLLM_VERSION=0.9.1
export VLLM_ENABLE_FUSED_EXPERTS_ALLGATHER_EP=1
nohup python -m vllm.entrypoints.openai.api_server --model=/mnt/deepseek/DeepSeek-R1-W8A8-VLLM \
    --served-model-name auto \
    --quantization ascend \
    --trust-remote-code \
    --distributed-executor-backend=mp \
    --port 8006 \
    -tp=4 \
    -dp=4 \
    --max-num-seqs 24 \
    --max-model-len 32768 \
    --max-num-batched-tokens 32768 \
    --block-size 128 \
    --no-enable-prefix-caching \
    --enable_expert_parallel \
    --additional-config '{"torchair_graph_config":{"enabled":true,"use_cached_graph":true,"graph_batch_sizes":[24]},"ascend_scheduler_config":{"enabled":true},"expert_tensor_parallel_size":16}' \
    --gpu-memory-utilization 0.96 &> run.log &
disown

- vLLM version: v0.9.2
- vLLM main: https://github.com/vllm-project/vllm/commit/e7e3e6d2636f6cd012c7ffeff773b20b3c90b958

### CODE REVIEW

**Comment by wangxiyuan** [comment]
please rebase to fix the merge conflict if this PR is still needed.

File: vllm_ascend/ops/fused_moe.py [hunk 1]
```diff
@@ -1311,20 +1311,25 @@ def forward(self,
...
                                                      tp_size,
                                                      dim=0)
-            chunk_router_logits = torch.tensor_split(router_logits,
-                                                     tp_size,
-                                                     dim=0)
             tp_rank = get_tensor_model_parallel_rank()
             hidden_states = chunk_hidden_states[tp_rank]
-            router_logits = chunk_router_logits[tp_rank]
+            if not self.rm_router_logits:
+                if num_tokens < tp_size:
```
Comment (momo609) [inline vllm_ascend/ops/fused_moe.py:1320]: This f statement's condition check can be directly merged with the padding check for num_token above.

**Comment by wangxiyuan** [comment]
No update for long time. I'll close it. Feel free to createa new one if you're still working on this.
